examples/addsvc/client/grpc/factory.go

```diff
@@ -7,33 +7,45 @@ import (
 
 	"github.com/go-kit/kit/endpoint"
 	"github.com/go-kit/kit/examples/addsvc/pb"
+	"github.com/go-kit/kit/loadbalancer"
+	"github.com/go-kit/kit/log"
+	kitot "github.com/go-kit/kit/tracing/opentracing"
 	grpctransport "github.com/go-kit/kit/transport/grpc"
+	"github.com/opentracing/opentracing-go"
 )
 
-// SumEndpointFactory transforms GRPC host:port strings into Endpoints that call the Sum method on a GRPC server
+// MakeSumEndpointFactory returns a loadbalancer.Factory that transforms GRPC
+// host:port strings into Endpoints that call the Sum method on a GRPC server
 // at that address.
-func SumEndpointFactory(instance string) (endpoint.Endpoint, io.Closer, error) {
-	cc, err := grpc.Dial(instance, grpc.WithInsecure())
-	return grpctransport.NewClient(
-		cc,
-		"Add",
-		"Sum",
-		encodeSumRequest,
-		decodeSumResponse,
-		pb.SumReply{},
-	).Endpoint(), cc, err
+func MakeSumEndpointFactory(tracer opentracing.Tracer, tracingLogger log.Logger) loadbalancer.Factory {
```

Comment:
Yeah, I think this is a reasonable-enough idiom for threading the logger through. My only other thought would be to define a Go kit tracer struct with the opentracing.Tracer + log.Logger together, but I think that's overkill at this point.